packages/libs/ledger/src/ledger.ts

```diff
@@ -0,0 +1,165 @@
+import { LedgerTransport, StatusCodes } from 'ledger-transport-hid';
+import { KADENA_PATH, LEDGER_VENDOR_ID } from './constants';
+import { TransactionParams, createTransaction } from './transaction';
+import { arrayBufferToHex, concatUint8Array, convertDecimal } from './utils';
+
+const NOT_SUPPORTED_ERROR =
+  'Your browser does not support connecting to hardware devices.';
+
+export class KadenaLedger {
+  transport: LedgerTransport;
+
+  constructor(device: HIDDevice) {
+    if (!KadenaLedger.isSupported()) {
+      throw new Error(NOT_SUPPORTED_ERROR);
+    }
+
+    this.transport = new LedgerTransport(device);
+  }
+
+  async openApp() {
+    await this.transport.send(
+      0xe0,
+      0xd8,
+      0x00,
+      0x00,
+      new TextEncoder().encode('Kadena'),
+      [StatusCodes.OK, 0x6e01], // 0x6e01 = already open
+    );
+    // Opening the app makes the device reconnect, wait a moment for this
+    await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
that's should be done more smarter, the reconnecting' isn't. there a event available we can listen to, this doesn't seem very error prone